src/enki_checks.py

```diff
@@ -1,8 +1,29 @@
 import re
+import itertools
 
 from enki_msg import Report
 from enki_regex import Regexes, Tags
 
+# standalone test to run on code blocks
+def sudo_check(
+    stripped_file: str,
+    report: Report,
+    file_path: str) -> None:
+    """Checks if sudo access is mentioned in documentation"""
+    code_block_dots = Regexes.CODE_BLOCK_DOTS.findall(stripped_file)
+    code_block_dashes = Regexes.CODE_BLOCK_DASHES.findall(stripped_file)
+    code_block_two_dashes = Regexes.CODE_BLOCK_TWO_DASHES.findall(stripped_file)
+
+    code_block = code_block_dots + code_block_dashes + code_block_two_dashes
+
+    # regex for root access  "# ", "~]# ", "*# ", " # ", "[root@", "[command]`# ", "#{nbsp}", "pass:quotes[#"
+    for item in ["$ sudo ", "$ *sudo ", "su -", "*su -"]:
+        for block in code_block:
+            if [b for b in block if b.startswith(item)]:
```

Comment:
If I understand this correctly, you're checking if the whole code block starts with a variation of sudo. But what if the code block contains multiple commands across its lines, and sudo is somewhere in the middle?

I'd recommend splitting the code blocks by lines and checking the start of each line separately.

Reply:
I'm checking lines, that's where `b for b in block` comes from

Reply:
> I'd recommend splitting the code blocks by lines and checking the start of each line separately.

I'm checking lines, that's where `b for b in block` comes from

Reply:
Ah, okay, I misread then. This could be clearer:

```python
            lines = block.splitlines()
            if [line for line in lines if line.startswith(item)]:
```

Or to make it more explicit that you're checking if any of the lines matches the condition, and you don't care about _all_ the lines:

```python
            lines = block.splitlines()
            if any(map(lambda line: line.startswith(item), lines))
```